plugins/widget/plugin.js

```diff
@@ -1066,17 +1064,11 @@
 		 * @returns {Boolean} Whether the style is applied to this widget.
 		 */
 		checkStyleActive: function( style ) {
-			var classes = getStyleClasses( style ),
-				cl;
-
-			if ( !classes )
-				return false;
+			style = CKEDITOR.tools.copy( style );
```

Comment:
`CKEDITOR.tools.copy` does a shallow copy so it is not completely safe to use it (as changing copy may change source object). There are two solutions here:

* Use `CKEDITOR.tools.clone` which does deep copy.
* Create new style using definition from source one: `new CKEDITOR.style( CKEDITOR.tools.clone( style._.definition ) )` (still we need to clone the definition).

I'm for the second one as cloning whole `CKEDITOR.style` object may be expensive. And if you look on [`CKEDITOR.style` constructor](https://github.com/ckeditor/ckeditor-dev/blob/9409e28dadf1564ec35987bcd888dfe6df23ef0d/core/style.js#L142) it doesn't perform anything fancy or computational heavy so it should be faster than cloning whole object.